plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -333,16 +333,19 @@ def forward(self, image, scales=None):
         )
 
         representation_vector = self.convert_to_vector().unsqueeze(0).unsqueeze(0)
-
+        
         if scales is not None:
-            ind = torch.tensor(
-                [
-                    i
-                    for i, s in enumerate(self.representation_scales)
-                    if s in scales
-                ]
-            ).to(image.device)
-            return representation_vector.index_select(-1, ind)
+            for ii,s in enumerate(self.representation_scales):
+                if s not in scales:
+                    representation_vector[:,:,ii]=0
+            # ind = torch.tensor(
+            #     [
+            #         i
+            #         for i, s in enumerate(self.representation_scales)
+            #         if s in scales
+            #     ]
+            # ).to(image.device)
+            # return representation_vector.index_select(-1, ind)
```

Comment:
What's going on here? It looks like you changed the behavior with `scales` so that you zero values out instead of just returning the relevant values. Why'd you make that change? (and you should delete the old way, not just comment it out!)

Reply:
There's an issue with converting to and from a dictionary if you actually remove the values.  I don't remember all the details.  I have removed the comments.